lacci/lib/shoes/drawable.rb

```diff
@@ -242,17 +242,15 @@ def shoes_style_name?(name)
       # not kept long, and used up when used once.
 
       def with_current_app(app)
-        old_cur_app = @current_app
-        @current_app = app
-        ret = yield
-        @current_app = old_cur_app
-        ret
+        old_app = Thread.current[:shoes_app]
```

Comment:
Interesting. Were you doing something that needed this? This is going to change multithreaded behaviour... Which is maybe fine? This way could be better if you have multiple threads and you assume each thread will be running a different app. It will fail if multiple threads ever need to use the *same* app.

Reply:
I was a bit deep when I put this together. Let me revisit this code and either explain myself or revert. I'd mentioned in the discord but some of my contributions today were a bit aggressive which increases the risk that I shipped a change not pivotal to my feature.